ethcore/src/executive.rs

```diff
@@ -573,13 +573,16 @@ impl<'a, B: 'a + StateBackend> Executive<'a, B> {
 		vm_tracer.done_subtrace(subvmtracer);
 
 		match res {
-			Ok(ref res) if res.apply_state => tracer.trace_create(
-				trace_info,
-				gas - res.gas_left,
-				trace_output.map(|data| output.as_ref().map(|out| out.to_vec()).unwrap_or(data)),
-				created,
-				subtracer.drain()
-			),
+			Ok(ref res) if res.apply_state => {
+				substate.contracts_created.push(created.clone());
```

Comment:
Not really sure about this change. This was a hack initially. As understand the initial design `Substate` is supposed to hold changes made in the context of execution. While creating a contract a result of creation transaction happens after that. There should be a better way to get contract address after execution.